babel/numbers.py

```diff
@@ -385,6 +384,205 @@ def format_scientific(number, format=None, locale=LC_NUMERIC):
     return pattern.apply(number, locale)
 
 
+def _find_unit_pattern(unit_id, length='long', locale=LC_NUMERIC):
+    """
+    Expand an unit and length into a qualified form.
+
+    Known units can be found in the CLDR Unit Validity XML file:
+    http://unicode.org/repos/cldr/tags/latest/common/validity/unit.xml
+
+    >>> _find_unit_pattern("radian", locale="en")
+    'angle-radian:long'
+
+    Unknown values will return None.
+
+    >>> _find_unit_pattern("horse", locale="en")
+
+    :param unit_id: the code of a measurement unit.
+    :param length: The length for the formatting patterns (long, short or narrow)
+    :return: A key to the `unit_patterns` mapping, or None.
+    """
+    locale = Locale.parse(locale)
+    pattern = "%s:%s" % (unit_id.split(":", 1)[0], length)
+    if pattern in locale.unit_patterns:
+        return pattern
+    for unit_pattern in sorted(locale.unit_patterns, key=len):
+        if unit_pattern.endswith(pattern):
```

Comment:
Hmm.. What's an example that you would expect to satisfy this condition?

I would have expected a fallback condition along the lines of: `unit_pattern.startswith(unit_id + ':')`